retrofit-converters/wire/src/main/java/retrofit/converter/WireConverter.java

```diff
@@ -36,31 +36,24 @@ public WireConverter(Wire wire) {
       throw new IllegalArgumentException("Expected a proto message but was " + c.getName());
     }
 
-    if (!MIME_TYPE.equalsIgnoreCase(body.mimeType())) {
-      throw new IllegalStateException("Expected a proto but was: " + body.mimeType());
```

Comment:
.... although here I think you're making so many forward-progress but incompatible changes that it doesn't really matter. You aren't working to a .1 release, this is a major release.